pkg/scheduler/factory_test.go

```diff
@@ -67,90 +66,233 @@ func TestCreate(t *testing.T) {
+			configData: []byte(`{
+				"kind" : "Policy",
+				"apiVersion" : "v1",
+				"predicates" : [
+					{"name" : "TestZoneAffinity", "argument" : {"serviceAffinity" : {"labels" : ["zone"]}}},
+					{"name" : "TestRequireZone", "argument" : {"labelsPresence" : {"labels" : ["zone"], "presence" : true}}},
+					{"name" : "PodFitsResources"},
+					{"name" : "PodFitsHostPorts"}
+				],
+				"priorities" : [
+					{"name" : "RackSpread", "weight" : 3, "argument" : {"serviceAntiAffinity" : {"label" : "rack"}}},
+					{"name" : "NodeAffinityPriority", "weight" : 2},
+					{"name" : "ImageLocalityPriority", "weight" : 1},
+					{"name" : "InterPodAffinityPriority", "weight" : 1}
+				],
+				"hardPodAffinitySymmetricWeight" : 10
+			}`),
+			wantPluginConfig: []schedulerapi.PluginConfig{
+				{
+					Name: nodelabel.Name,
+					Args: &schedulerapi.NodeLabelArgs{
+						PresentLabels: []string{"zone"},
+					},
+				},
+				{
+					Name: serviceaffinity.Name,
+					Args: &schedulerapi.ServiceAffinityArgs{
+						AffinityLabels:               []string{"zone"},
+						AntiAffinityLabelsPreference: []string{"rack"},
+					},
+				},
+				{
+					Name: interpodaffinity.Name,
+					Args: &schedulerapi.InterPodAffinityArgs{
+						HardPodAffinityWeight: 10,
+					},
+				},
+			},
+			wantPlugin: &schedulerapi.Plugins{
+				QueueSort: &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{{Name: "PrioritySort"}}},
+				PreFilter: &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{
+					{Name: "NodePorts"},
+					{Name: "NodeResourcesFit"},
+					{Name: "ServiceAffinity"},
+				}},
+				Filter: &schedulerapi.PluginSet{
+					Enabled: []schedulerapi.Plugin{
+						{Name: "NodeUnschedulable"},
+						{Name: "NodePorts"},
+						{Name: "NodeResourcesFit"},
+						{Name: "TaintToleration"},
+						{Name: "NodeLabel"},
+						{Name: "ServiceAffinity"},
+					},
+				},
+				PostFilter: &schedulerapi.PluginSet{},
+				PreScore:   &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{{Name: "InterPodAffinity"}}},
+				Score: &schedulerapi.PluginSet{
+					Enabled: []schedulerapi.Plugin{
+						{Name: "ImageLocality", Weight: 1},
+						{Name: "InterPodAffinity", Weight: 1},
+						{Name: "NodeAffinity", Weight: 2},
+						{Name: "ServiceAffinity", Weight: 3},
+					},
+				},
+				Reserve:  &schedulerapi.PluginSet{},
+				Permit:   &schedulerapi.PluginSet{},
+				PreBind:  &schedulerapi.PluginSet{},
+				Bind:     &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{{Name: "DefaultBinder"}}},
+				PostBind: &schedulerapi.PluginSet{},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			client := fake.NewSimpleClientset()
+
+			informerFactory := informers.NewSharedInformerFactory(client, 0)
+			recorderFactory := profile.NewRecorderFactory(events.NewBroadcaster(&events.EventSinkImpl{Interface: client.EventsV1()}))
+
+			_, err := New(
+				client,
+				informerFactory,
+				recorderFactory,
```

Comment:
Have you tried setting these 2 to nil?

Reply:
It raises and exception for both, *informerFactory* is used for the service affinity, and the recorder is used when creating a new Profile.